tests/GitHub/App/Token/PermissionsSpec.hs

```diff
@@ -0,0 +1,21 @@
+module GitHub.App.Token.PermissionsSpec
+  ( spec
+  ) where
+
+import GitHub.App.Token.Prelude
+
+import Data.Aeson
+import GitHub.App.Token.Permissions
+import Test.Hspec
+
+spec :: Spec
+spec = do
+  describe "Semigroup" $ do
+    it "resolves duplicates by taking higher permission" $ do
+      encode (checks Read <> checks Write)
+        `shouldBe` "{\"checks\":\"write\"}"
```

Comment:
Interesting example of doing something awkward to avoid testing internals.

Reply:
Maybe this would be better,

```hs
encode (checks Read <> checks Write) `shouldBe` encode (checks Write)
```

Still awkward, but doesn't depend on `encode`'s functionality.

Reply:
Can't you rely on `instance Eq Permissions`? Or what about exposing a getter to retrieve the `Permission`?